hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRolloutResource.java

```diff
@@ -263,6 +264,11 @@ public ResponseEntity<MgmtRolloutGroupResponseBody> getRolloutGroup(@PathVariabl
 
         final RolloutGroup rolloutGroup = rolloutGroupManagement.getWithDetailedStatus(groupId)
                 .orElseThrow(() -> new EntityNotFoundException(RolloutGroup.class, rolloutId));
+
+        if (!Objects.equals(rolloutId, rolloutGroup.getRollout().getId())) {
+            throw new EntityNotFoundException("The provided rolloutId does not match with the rolloutId in the deployment group.");
```

Comment:
I would not expose this information that the group does exist and that it belongs to a different rollout. I'd recommend to return a plain 404 with a corresponding message ("Group XYZ not found").